jbpm-flow/src/main/java/org/jbpm/process/core/timer/impl/ThreadPoolSchedulerService.java

```diff
@@ -91,21 +91,15 @@ public void shutdown() {
     @Override
     public JobHandle scheduleJob(Job job, JobContext ctx, Trigger trigger) {
 
+        long id = idCounter.getAndIncrement(); 
         Date date = trigger.hasNextFireTime();
         if ( date != null ) {
-            String jobname = null;
-            if (ctx instanceof ProcessJobContext) {
-                ProcessJobContext processCtx = (ProcessJobContext) ctx;
-                jobname = processCtx.getSessionId() + "-" + processCtx.getProcessInstanceId() + "-" + processCtx.getTimer().getId();
-                if (processCtx instanceof StartProcessJobContext) {
-                    jobname = "StartProcess-"+((StartProcessJobContext) processCtx).getProcessId()+ "-" + processCtx.getTimer().getId();
-                }
-                if (activeTimer.containsKey(jobname)) {
-                    return activeTimer.get(jobname);
-                }
-            
+            String jobname = JobNameHelper.getJobName(ctx, id);
+            if (activeTimer.containsKey(jobname)) {
+                return activeTimer.get(jobname);
             }
```

Comment:
@mswiderski So until now if the `JobContext` was other than  `ProcessJobContext` the job name was left null and such job wasn't put in the map of active timers, right? So for example for `TaskDeadlineJobContext` (instance of NamedJobContext) we might have ended up with the same timer scheduled more than once? So was that actually a bug?

As right now we will create a job name for every possible JobContext using `JobNameHelper` and that also means all timers will be remembered.